centaur/src/main/resources/standardTestCases/drs_tests/drs_basename.wdl

```diff
@@ -0,0 +1,52 @@
+version 1.0
+
+workflow drs_basename {
+    input {
+        File file
+    }
+
+    String workflow_level_basename = basename(file)
+
+    call pump_up_the_base { input:
+        file = file,
+        pre_basenamed = workflow_level_basename,
+        basednamed_at_call_site = basename(file)
+    }
+
+    output {
+        Array[String] basenames = pump_up_the_base.basenames
+    }
+}
+
+task pump_up_the_base {
+    input {
+        File file
+        String pre_basenamed
+        String basednamed_at_call_site
+    }
+
+    String task_level_basename = basename(file)
+
+    command <<<
+        echo ~{pre_basenamed}
+        echo ~{basednamed_at_call_site}
+        echo ~{task_level_basename}
+        echo ~{basename(file)}
+        # Also validate the two-parameter version of the function with a DRS input for file extension replacements:
+        echo ~{basename(file, ".fastq.gz")}.foo.bar
+    >>>
+
+    output {
+        Array[String] basenames = read_lines(stdout())
+    }
+
+    runtime {
+        docker: "ubuntu"
+        backend: "papi-v2-usa"
+    }
+
+    parameter_meta {
+        # We don't actually want to use this file, just play around with its basename:
+        file: { localization_optional: true }
```

Comment:
is this being honored for DRS?

Reply:
My notes (from back in March) say that there is a Cromwell feature flag (currently off in Terra) that will respect this iff Martha says that there isn't a bond provider for the object, in which case Cromwell will translate the DRS URI into a native GS URI. The only data provider that I know of that this will work for is TDR. If the feature flag is on and there is a bond provider for an object, `localization_optional` will be ignored for that object and localization will always be done.

We might want to tweak this behavior in the future so that we return a signed URL instead to support tools that can make HTTP range requests.

Reply:
Specifically, I think the flag is `filesystems.drs.config.martha.preresolve` (not 100% sure about the config path) which I imagine is not turned on for this test. Therefore, it's probably best to remove this `localization_optional` to avoid confusion.

Now that I think about it, I wonder if `basename` behaves correctly when `preresolve` is `true`. 🤔  Probably worth testing that if there isn't already a test for it. Maybe adding some `basename` checks to `drs_usa_jdr_preresolve.test` would be appropriate.

Reply:
Welp it looks like `FileLocalization.md` and `DataRepositoryService.md` need some updates on this topic...

Thanks for checking your notes! It looks like the auto-conversion from a `DrsPath` to a `GcsPath` is dependent on [preresolve](https://github.com/broadinstitute/cromwell/blob/6621d30ec56af65a40d9157df7a3fcca2103bf65/filesystems/drs/src/main/scala/cromwell/filesystems/drs/DrsPathBuilder.scala#L30) and the [absence of a Bond provider](https://github.com/broadinstitute/cromwell/blob/6621d30ec56af65a40d9157df7a3fcca2103bf65/filesystems/drs/src/main/scala/cromwell/filesystems/drs/DrsResolver.scala#L51) via [getSimpleGsUri](https://github.com/broadinstitute/cromwell/blob/6621d30ec56af65a40d9157df7a3fcca2103bf65/filesystems/drs/src/main/scala/cromwell/filesystems/drs/DrsPathBuilder.scala#L64).

But it also seems that there's a path for `localization_optional` to [skip localization](https://github.com/broadinstitute/cromwell/blob/develop/supportedBackends/google/pipelines/common/src/main/scala/cromwell/backend/google/pipelines/common/PipelinesApiAsyncBackendJobExecutionActor.scala#L902) independent of preresolve but with the same no-Bond and yes-URI conditions. I believe those should be the conditions for this particular test since it's talking to TDR.

Somewhat analogous to Brian's question, I'm wondering if this works on a non-TDR repo.

Reply:
Also might be worth mentioning in the CHANGELOG.md as a bugfix (and with any caveats if this doesn't work with Bond-ed providers or signed URLs).

Reply:
Update: This was tested by @mcovarr against non-TDR and worked as expected. Perhaps thanks to changing to get the name directly from the response rather than indirectly through fetching a URL and modifying it?